### PR TITLE
feat: analytic events for new search bar container

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Feat

This PR solves [FX-4780]

Tasks
- [x] New events for tracking switching between pills.
- [x] Replace old “deprecated” events with new ones. 
- [x] Make sure that all events are tracked correctly

[FX-4780]: https://artsyproduct.atlassian.net/browse/FX-4780?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Search/NewSearch/Mobile/Overlay.tsx [hunk 1]
```diff
@@ -44,11 +44,12 @@ export const Overlay: FC<OverlayProps> = ({ viewer, relay, onClose }) => {
     tracking.trackEvent({
       action_type:
         options.length > 0
-          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
-          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+          ? ActionType.selectedSearchSuggestionQuickNavigationItem
```
Comment (dblandin) [inline src/Components/Search/NewSearch/Mobile/Overlay.tsx:47]: **suggestion**: I think this "QuickNavigationItem" refers to the (Artworks) or (Auction Results) links in Artist results and not a search result/suggestion itself. We might need to stick with these deprecated action types as I think they more accurately reflect what this event is meant to track.

File: src/Components/Search/NewSearch/NewSearchBarFooter.tsx [hunk 2]
```diff
@@ -2,26 +2,30 @@ import ChevronRightIcon from "@artsy/icons/ChevronRightIcon"
...
   index,
+  selectedPill,
 }) => {
   const tracking = useTracking()
 
   const handleClick = () => {
     tracking.trackEvent({
-      action_type: DeprecatedSchema.ActionType.SelectedItemFromSearch,
+      action_type: ActionType.selectedItemFromSearch,
+      context_module: selectedPill.analyticsContextModule,
```
Comment (dblandin) [inline src/Components/Search/NewSearch/NewSearchBarFooter.tsx:28]: Nice 💯

**Review by dblandin — COMMENTED** [review]
Looking good! Left a few questions/comments.

File: src/Components/Search/NewSearch/Mobile/Overlay.tsx [hunk 3]
```diff
@@ -44,11 +44,12 @@ export const Overlay: FC<OverlayProps> = ({ viewer, relay, onClose }) => {
...
+          : ActionType.searchedWithNoResults,
+      context_module: selectedPill.analyticsContextModule,
       query: value,
     })
-  }, [fetchCounter])
+  }, [fetchCounter, options])
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/Components/Search/NewSearch/Mobile/Overlay.tsx:52]: **blocking**: every time a component is re-rendered, `options` will return a new array and a **new** reference type. For this reason, `useUpdateEffect` will be called **every** time when component is re-rendered and the event will be tracked accordingly

[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/Components/Search/NewSearch/Mobile/Overlay.tsx:52]: and +1 to the comment from Devon

**Review by dimatretyak — CHANGES_REQUESTED** [review]

**Review by dimatretyak — APPROVED** [review]

**Review by anandaroop — APPROVED** [review]

**Comment by rajsam003** [comment]
@dblandin Thanks for rebasing it

**Comment by dblandin** [comment]
No problem! I noticed the Cypress test failure and then [this PR](https://github.com/artsy/force/pull/12466) from @anandaroop so hopefully the build will pass now 🙏🏼

**Pull request merged**